eth/tracers/api_blocktrace.go

```diff
@@ -30,20 +32,51 @@ func (api *API) GetBlockTraceByNumberOrHash(ctx context.Context, blockNrOrHash r
 	if block.NumberU64() == 0 {
 		return nil, errors.New("genesis is not traceable")
 	}
-	if config == nil {
-		config = &TraceConfig{
-			LogConfig: &vm.LogConfig{
-				EnableMemory:     false,
-				EnableReturnData: true,
-			},
+
+	env, err := api.createTraceEnv(ctx, config, nil, block)
+	if err != nil {
+		return nil, err
+	}
+
+	return env.GetBlockTrace(block)
+}
+
+func (api *API) GetBlockTraceForCallAt(ctx context.Context, args ethapi.TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, config *TraceCallConfig) (*types.BlockTrace, error) {
+	// Try to retrieve the specified block
+	var (
+		err   error
+		block *types.Block
+	)
+	if hash, ok := blockNrOrHash.Hash(); ok {
+		block, err = api.blockByHash(ctx, hash)
+	} else if number, ok := blockNrOrHash.Number(); ok {
+		block, err = api.blockByNumber(ctx, number)
+	} else {
+		return nil, errors.New("invalid arguments; neither block nor hash specified")
+	}
+	if err != nil {
+		return nil, err
+	}
+	if block.NumberU64() == 0 {
+		return nil, errors.New("genesis is not traceable")
+	}
+
+	block = types.NewBlockWithHeader(block.Header()).WithBody([]*types.Transaction{args.ToTransaction()}, nil)
```

Comment:
So we actually replay and trace the transaction on top of `block.parent`'s state, right?

Reply:
yes, see `(api *API) createTraceEnv`.

we apply a mock block, containing the given tx/call, on top of the parent block

Reply:
so your point is, the API name can be confusing? I will agree.

Reply:
what about ·scroll_getBlockTraceForCallAtParentBlock`?

Reply:
I think as long as the behaviour is consistent with `debug_traceCall` then it's okay. Btw you can also consider naming it `scroll_traceCall`.

Reply:
then it's different.


`debug_traceCall` traces at the given block
but this new API traces at the parent block